src/main/java/org/mskcc/oncokb/curation/importer/CanonicalTranscriptFlagImporter.java

```diff
@@ -0,0 +1,120 @@
+package org.mskcc.oncokb.curation.importer;
+
+import static org.mskcc.oncokb.curation.util.FileUtils.readDelimitedLinesStream;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.*;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.mskcc.oncokb.curation.domain.Flag;
+import org.mskcc.oncokb.curation.domain.Transcript;
+import org.mskcc.oncokb.curation.domain.enumeration.FlagType;
+import org.mskcc.oncokb.curation.domain.enumeration.ReferenceGenome;
+import org.mskcc.oncokb.curation.service.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+/**
+ * To backfill the OncoKB canonical transcripts
+ */
+
+@Component
+public class CanonicalTranscriptFlagImporter {
+
+    private final Logger log = LoggerFactory.getLogger(CanonicalTranscriptFlagImporter.class);
+
+    final String ONCOKB_GENE_LIST_FILE_PATH = "data/oncokb_core/cancer_gene_list_v4_23.tsv";
+
+    final GeneService geneService;
+    final TranscriptService transcriptService;
+    final FlagService flagService;
+
+    public CanonicalTranscriptFlagImporter(
+        GeneService geneService,
+        GenomeNexusService genomeNexusService,
+        EnsemblGeneService ensemblGeneService,
+        TranscriptService transcriptService,
+        FlagService flagService,
+        InfoService infoService,
+        MainService mainService
+    ) {
+        this.geneService = geneService;
+        this.transcriptService = transcriptService;
+        this.flagService = flagService;
+    }
+
+    public void importCanonicalTranscripts() {
```

Comment:
Is this basically a data migration? In other words, does this only need to be run with a new data release?

Reply:
This can be ran with a new data release if needed, but I would just wait for the feature oncokb/oncokb-pipeline#634.
The main purpose is to update the transcript information we have saved in oncokb_curation with the information in oncokb_core.